src/wui/fieldaction.cc

```diff
@@ -853,6 +860,16 @@ void FieldActionWindow::act_abort_buildwaterway() {
 	reset_mouse_and_die();
 }
 
+void FieldActionWindow::act_goto_buildroadstart() {
+	if (!ibase().in_road_building_mode()) {
+		return;
+	}
+	ibase().map_view()->scroll_to_field((SDL_GetModState() & KMOD_CTRL) != 0 ?
+                                          ibase().get_build_road_end() :
+                                          ibase().get_build_road_start(),
+	                                    MapView::Transition::Smooth);
```

Comment:
we should reset and let the mouse die here as well, as the window is perfectly covering the actual road cursor indicated by the steepnees indicators. 
rest looks fine